tests/qnn/test_keras.py

```diff
@@ -826,20 +845,22 @@ def circuit(x, weights):
         qml.RY(weights[1], wires=1)
         return qml.probs(op=qml.PauliZ(1))
 
-    KerasLayer.set_input_argument("x")
-    layer = KerasLayer(circuit, weight_shapes={"weights": (2,)}, output_dim=(2,))
-    layer.build((None, 2))
+    try:
+        KerasLayer.set_input_argument("x")
+        layer = KerasLayer(circuit, weight_shapes={"weights": (2,)}, output_dim=(2,))
+        layer.build((None, 2))
 
-    x = tf.constant(np.random.uniform(0, 1, (10, 4)))
-    res = layer(x)
+        x = tf.constant(np.random.uniform(0, 1, (10, 4)))
+        res = layer(x)
 
-    assert res.shape == (10, 2)
+        assert res.shape == (10, 2)
 
-    for x_, r in zip(x, res):
-        assert qml.math.allclose(r, circuit(x_, layer.qnode_weights["weights"]), atol=tol)
+        for x_, r in zip(x, res):
+            assert qml.math.allclose(r, circuit(x_, layer.qnode_weights["weights"]), atol=tol)
 
-    # reset back to the old name
-    KerasLayer.set_input_argument("inputs")
+    finally:
+        # reset back to the old name
+        KerasLayer.set_input_argument("inputs")
```

Comment:
If we defined that function earlier for reseting the Keras inputs, do we want to make it reusable here as a fixture, instead of explicitly defining similar functionality?

Reply:
You are right this one should be safe to remove

Reply:
Trying here https://github.com/PennyLaneAI/pennylane/pull/7039/commits/4beefa1c80048b76862ad0ca64150b189e99c1af

Reply:
CI passed